### PR TITLE
Introduce version for spotbugs annotations for re-use in jenkins bom

### PR DESCRIPTION
**Pull request by StefanSpieker**
Introduce version for spotbugs annotations for re-use in jenkins bom.
Follow-Up of: https://github.com/jenkinsci/jenkins/pull/4689

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,7 +147,7 @@
       <dependency>
         <groupId>com.github.spotbugs</groupId>
         <artifactId>spotbugs-annotations</artifactId>
-        <version>4.0.1</version>
+        <version>${spotbus-annotations.version}</version>
```
Comment (MarkEWaite) [inline pom.xml:150]: Typo?
```suggestion
        <version>${spotbugs-annotations.version}</version>
```

File: pom.xml [hunk 2]
```diff
@@ -897,6 +897,7 @@
     <java.level.test>${java.level}</java.level.test>
 
     <spotbugs-maven-plugin.version>4.0.0</spotbugs-maven-plugin.version>
+    <spotbus-annotations.version>4.0.1</spotbus-annotations.version>
```
Comment (MarkEWaite) [inline pom.xml:900]: Could we use 4.0.2 that was released in April 2020?

Same spelling change (hoping that the spelling is correct where the property is defined)
```suggestion
    <spotbugs-annotations.version>4.0.1</spotbugs-annotations.version>
```

**Comment by StefanSpieker** [comment]
Thanks a lot for fixing the typeo.

**Comment by MarkEWaite** [comment]
Dependabot proposal in #97 to update to 4.0.2

File: pom.xml [hunk 3]
```diff
@@ -897,6 +897,7 @@
     <java.level.test>${java.level}</java.level.test>
 
     <spotbugs-maven-plugin.version>4.0.0</spotbugs-maven-plugin.version>
+    <spotbugs-annotations.version>4.0.1</spotbugs-annotations.version>
```
Comment (timja) [inline pom.xml:900]:
```suggestion
    <spotbugs-annotations.version>4.0.2</spotbugs-annotations.version>
```

Comment (timja) [reply pom.xml:900]: or we can merge the dependabot PR and rebase here

Comment (timja) [reply pom.xml:900]: or let dependabot rebase

**Pull request merged**